src/hooks/useCioAutocomplete.ts

```diff
@@ -50,14 +55,27 @@ const useCioAutocomplete = (options: UseCioAutocompleteOptions) => {
   );
 
   // Get dropdown items array from active sections (autocomplete + recommendations + custom)
-  const items = useItems(activeSectionsWithData);
-
-  const downshift = useDownShift({ setQuery, items, onSubmit, cioClient, previousQuery });
-  const { isOpen, getMenuProps, getLabelProps, openMenu, closeMenu, highlightedIndex } = downshift;
+  const items = useMemo(
+    () => getItemsForActiveSections(activeSectionsWithData),
+    [activeSectionsWithData]
```

Comment:
@mocca102, are we memoizing `activeSectionsWithData`?

in order for this dependency array and useMemo hook to  behave as intended, we'll want to make sure that we are not just passing a reference to an object or array that will change on every re-render. 

dependency arrays should only be passed:
 - primitives
 - memoized objs/arrays
 - objs/arrays that are use the same reference across renders (e.g. variables managed by useState hook)

for reference, more on this topic [here](https://react.dev/reference/react/useMemo#memoizing-a-dependency-of-another-hook)

Reply:
I believe we meet the third criteria here 😃 

> objs/arrays that are use the same reference across renders (e.g. variables managed by useState hook)

`activeSectionsWithData` is returned from `useSections` hook and inside it `activeSectionsWithData` is a variable managed by `useState` hook